test/tests/val-2e/comparison_val-2e.py

```diff
@@ -0,0 +1,518 @@
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib import gridspec
+import pandas as pd
+import os
+
+# Changes working directory to script directory (for consistent MooseDocs usage)
+script_folder = os.path.dirname(__file__)
+os.chdir(script_folder)
+
+#===============================================================================
+# Define methods
+
+def numerical_solution_on_experiment_input(experiment_input, tmap_input, tmap_output):
+    """interpolate numerical solution to the experimental time step
+
+    Args:
+        experiment_input (float, ndarray): experimental input data points
+        tmap_input (float, ndarray): numerical input data points
+        tmap_output (float, ndarray): numerical output data points
+
+    Returns:
+        float, ndarray: updated tmap_output based on the data points in experiment_input
+    """
+    new_tmap_output = np.zeros(len(experiment_input))
+    for i in range(len(experiment_input)):
+        left_limit = np.argwhere((np.diff(tmap_input < experiment_input[i])))[0][0]
+        right_limit = left_limit + 1
+        new_tmap_output[i] = (experiment_input[i] - tmap_input[left_limit]) / (tmap_input[right_limit] - tmap_input[left_limit]) * (tmap_output[right_limit] - tmap_output[left_limit]) + tmap_output[left_limit]
+    return new_tmap_output
+
+def read_csv_from_TMAP8(file_name, parameter_names):
+    """Read simulation data in csv files from TMAP8
+
+    Args:
+        file_name (string): the file name at simulation folder
+        parameter_names (list): the list of parameters extracted from csv files
+
+    Returns:
+        float, ndarray: the matrix keep the simulation results, first axis depended on len(parameter_names)
+    """
+    if "/TMAP8/doc/" in script_folder:     # if in documentation folder
```

Comment:
```suggestion
    if "/tmap8/doc/" in script_folder.lower():     # if in documentation folder
```
We now use this approach, which enable people to have their folder named either TMAP8 or tmap8.